agenthub/self_discover_agent/prompt.py

```diff
@@ -0,0 +1,214 @@
+
+USER:
+Based on the adapted reasoning modules implement a reasoning structure for your team members to follow step-by-step and arrive at the correct answer.
+
+## Task
+Recall from above the task: This SVG path element <path d="M 55.57,80.69 L 57.38,65.80 M 57.38,65.80 L 48.90,57.46 M 48.90,57.46 L 45.58,47.78 M 45.58,47.78 L 53.25,36.07 L 66.29,48.90 L 78.69,61.09 L 55.57,80.69"/> draws a:
+(A) circle (B) heptagon (C) hexagon (D) kite (E) line (F) octagon (G) pentagon(H) rectangle (I) sector (J) triangle
+
+## Constraints
+Do not solve the actual task yourself. Instead, you must plan the task by deriving a step-by-step reasoning structure, which guides your team to the correct answer.
+Your response MUST be in JSON format. It must be an object with the fields "{task_key}" and "{implement_state_key}":
+* {task_key}: this field must repeat the initial task from the user
+* {implement_state_key}: key-values pairs which represent the step-by-by plan.
+
+ASSISTANT:
+{{
+  "{task_key}": "This SVG path element <path d="M 55.57,80.69 L 57.38,65.80 M 57.38,65.80 L 48.90,57.46 M 48.90,57.46 L 45.58,47.78 M 45.58,47.78 L 53.25,36.07 L 66.29,48.90 L 78.69,61.09 L 55.57,80.69"/> draws a:
+(A) circle (B) heptagon (C) hexagon (D) kite (E) line (F) octagon (G) pentagon(H) rectangle (I) sector (J) triangle",
+  "{implement_state_key}": {{
+    "Step 1: Analyze the SVG Path Element": {{
+      "Step 1.1: Review the SVG Path Data": "Examine the coordinates and line segments defined in the SVG path element.",
+      "Step 1.2: Identify the Start and End Points": "Note the starting point and the ending point of the path to understand the sequence of drawing."
+    }},
+    "Step 2: Break Down the SVG Path into Parts": {{
+      "Step 2.1: Divide the Path into Segments": {{
+        "Step 2.1.1: Segment 1": "Analyze the coordinates and direction of the first line segment.",
+        "Step 2.1.2: Segment 2": "Analyze the coordinates and direction of the second line segment.",
+        "Step 2.1.3: Segment 3": "Analyze the coordinates and direction of the third line segment.",
+        "Step 2.1.4: Continue until all segments are analyzed": "Proceed with analyzing each segment sequentially."
+      }}
+    }},
+    "Step 3: Evaluate the Connections": {{
+      "Step 3.1: Determine the Connections Between Segments": "Check how each segment connects to the next one.",
+      "Step 3.2: Identify Closed Shapes or Open Paths": "Determine if the segments form a closed shape or an open path."
+    }},
+    "Step 4: Deduce the Overall Shape": {{
+      "Step 4.1: Compile the Analyzed Segments": "Combine the information from all segments to understand the overall structure.",
+      "Step 4.2: Match with Given Options": "Compare the deduced shape with the provided options (circle, heptagon, etc.) and identify the closest match."
+    }}
+  }}
+}}
+"""
+
+
+def get_prompt(
```

Comment:
On a second thought: the prompt here is probably a little bit too complex to support for the Jinja2 template in its current shape - we can defer that to future PR.